model/llm/llm_test.go

```diff
@@ -46,7 +46,7 @@ func TestModelLLMGenerateTestsForFile(t *testing.T) {
 
 			actualAssessment, actualError := llm.GenerateTestsForFile(tc.Language, temporaryPath, tc.SourceFilePath)
 			assert.NoError(t, actualError)
-			assert.Equal(t, tc.ExpectedAssessment, actualAssessment)
+			assert.Truef(t, tc.ExpectedAssessment.IsEqual(actualAssessment), "expected:%s\nactual:%s", tc.ExpectedAssessment, actualAssessment)
```

Comment:
Why are we changing this assert?

Reply:
Looks like reimplementing ".Equal"

Reply:
because `.Equal` also checks if keys exist but that should not matter for our assessments IMO, see https://github.com/symflower/eval-dev-quality/pull/39#discussion_r1570210755

Reply:
I see. can you introduce .Equal then with its own commit and add a helper to make that clear with a commit why we need it.